### PR TITLE
feat: Stub out NavigationStore

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Replace deprecated zustand APIs (`getState`, `setState`)
 - Refactor `editor.planx.uk/src/pages/FlowEditor/lib/store/index.ts` to make things a little clearer (opinions welcome!)
 - Stub out a basic `NavigationStore` which can hold and control upcoming section logic, along with very basic tests

### CODE REVIEW

**Comment by DafyddLlyr** [comment]
[02. Stub out Navigation Store](https://trello.com/c/23JEsVT8/2293-02-stub-out-navigation-store)

**Review by jessicamcinchak — APPROVED** [review]
works great! :checkered_flag: 

nit: noticed there's still a few zustand-related deprecation warnings in the console on the pizza, but not a blocker! 
Screenshot from 2023-02-17 13-43-22

File: editor.planx.uk/src/pages/FlowEditor/lib/__tests__/navigation.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { vanillaStore } from "../store";
```
[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply editor.planx.uk/src/pages/FlowEditor/lib/__tests__/navigation.test.ts:1]: This is intentionally testing that `navigationStore` values and functions are available in the combined store.

**Comment by DafyddLlyr** [comment]
> works great! 🏁
> 
> nit: noticed there's still a few zustand-related deprecation warnings in the console on the pizza, but not a blocker! Screenshot from 2023-02-17 13-43-22

There's one outstanding deprecation warning which I've spent a fair bit of time trying to resolve without much success - the docs weren't super clear on how to handle this. I'm going to go ahead and merge this and come back to this as a separate piece.

**Pull request merged**